examples/dendrogram.py

```diff
@@ -0,0 +1,150 @@
+#     3. Neither the name of the copyright holder nor the names of
+#        its contributors may be used to endorse or promote products
+#        derived from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY
+# DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+# ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+'''Moprhology Dendrogram Functions Example'''
+
+from neurom.view import common
+from neurom.analysis.morphmath import segment_length
+from neurom.core.tree import isegment
+from matplotlib.collections import LineCollection
+
+
+def get_transformed_position(segment, segments_dict, y_length):
+    '''Calculate Dendrogram Line Position
+    '''
+
+    start, end = segment
+
+    start_node_id = start.value[-2]
+
+    end_node_id = end.value[-2]
+
+    # The parent node is connected with the child
+    # line type:
+    # 0 straight segment
+    # 1 upward segment
+    # -1 downward segment
+    (x0, y0), line_type = segments_dict[start_node_id]
+
+    # increase horizontal dendrogram length by segment length
+    x1 = x0 + segment_length(segment)
+
+    # if the parent node is visited for the first time
+    # the dendrogram segment is drawn from below. If it
+    # is visited twice it is drawn from above
+    y1 = y0 + y_length[start_node_id] * line_type
+
+    # reduce the vertical y for subsequent children
+    y_length[end_node_id] = y_length[start_node_id] * (1. - 0.5 * abs(line_type))
+
+    # horizontal segment
+    segments = [((x0, y1), (x1, y1))]
+
+    # vertical segment
+    if line_type != 0:
+        segments.extend([((x0, y0), (x0, y1))])
+
+    # If the segment has children, the first child will be drawn
+    # from below. If no children the child will be a straight segment
+    segments_dict[end_node_id] = [(x1, y1), 1 - len(end.children)]
+
+    # the second branch will have diff direction
+    segments_dict[start_node_id][1] += 2
+
+    return segments
+
+
+def dendro_transform(tree_object):
+    '''Extract Dendrogram Lines and Diameters from tree structure
+    '''
+    root_id = tree_object.value[-2]
+
+    y_length = {root_id: 1.}
+
+    segments_dict = {tree_object.value[-2]: [(0., 0.), 0.]}
+
+    positions = []
+    diameters = []
+
+    for seg in isegment(tree_object):
```

Comment:
you could apply the value transformation here, instead of using x.value and so on further down the line. Something like

    for seg in val_iter(isegment(tree_object)):

        seg_diam = seg[0][3] + seg[1][3]